### PR TITLE
Replace goleak.VerifyTestMain with testutils.VerifyGoLeaks

### PR DESCRIPTION
**Pull request by Pushkarm029**
## Which problem is this PR solving?
- Part of #5006 

## Description of the changes
- Fixed Script to count `testutils.VerifyGoLeaks` as satisfactory.
- Replaced `goleak.VerifyTestMain` with `testutils.VerifyGoLeaks`.

## How was this change tested?
- `make test`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/agent/app/processors/thrift_processor_test.go [hunk 1]
```diff
@@ -33,9 +32,10 @@ import (
 	grpcrep "github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc"
 	"github.com/jaegertracing/jaeger/cmd/agent/app/servers"
 	"github.com/jaegertracing/jaeger/cmd/agent/app/servers/thriftudp"
-	"github.com/jaegertracing/jaeger/cmd/agent/app/testutils"
+	tu "github.com/jaegertracing/jaeger/cmd/agent/app/testutils"
```
Comment (yurishkuro) [inline cmd/agent/app/processors/thrift_processor_test.go:35]: instead of this invasive change, move goleak into package_test

File: pkg/gogocodec/codec_test.go [hunk 2]
```diff
@@ -11,7 +11,6 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
```
Comment (yurishkuro) [inline pkg/gogocodec/codec_test.go:14]: what's with the changes in this file?

**Review by yurishkuro — COMMENTED** [review]
you can also add the uber/goleak to disallowed packages in `.golangci.yml`, similar to `disallow-crossdock`

File: pkg/gogocodec/codec_test.go (hunk 2 above)
Comment (Pushkarm029) [reply pkg/gogocodec/codec_test.go:14]: I was resolving conflicts but forgot that I needed to remove this change.

File: .golangci.yml [hunk 3]
```diff
@@ -112,6 +112,13 @@ linters-settings:
             desc: "Do not refer to crossdock from other packages"
         files:
           - "!**/crossdock/**"
+      
+      disallow-uber/goleak:
+        deny:
+         - pkg: go.uber.org/goleak
+           desc: "Use github.com/jaegertracing/jaeger/pkg/testutils instead of go.uber.org/goleak"
+        files:
+          - "**_test.go**"
```
Comment (yurishkuro) [inline .golangci.yml:121]:
```suggestion
          - "**_test.go"
```

Comment (Pushkarm029) [reply .golangci.yml]: done

File: .golangci.yml [hunk 4]
```diff
@@ -112,6 +112,13 @@ linters-settings:
             desc: "Do not refer to crossdock from other packages"
         files:
           - "!**/crossdock/**"
+      
+      disallow-uber/goleak:
+        deny:
+         - pkg: go.uber.org/goleak
+           desc: "Use github.com/jaegertracing/jaeger/pkg/testutils instead of go.uber.org/goleak"
```
Comment (yurishkuro) [inline .golangci.yml:119]:
```suggestion
           desc: "Use github.com/jaegertracing/jaeger/pkg/testutils"
```

File: pkg/gogocodec/codec_test.go [hunk 5]
```diff
@@ -32,8 +32,8 @@ func TestCodecMarshallAndUnmarshall_jaeger_type(t *testing.T) {
 	s1 := &model.Span{OperationName: "foo", TraceID: model.NewTraceID(1, 2)}
 	data, err := c.Marshal(s1)
 	require.NoError(t, err)
-
 	s2 := &model.Span{}
+
```
Comment (yurishkuro) [inline pkg/gogocodec/codec_test.go:36]: why these changes?

File: .golangci.yml (hunk 4 above)
Comment (Pushkarm029) [reply .golangci.yml]: done

File: pkg/gogocodec/codec_test.go (hunk 5 above)
Comment (Pushkarm029) [reply pkg/gogocodec/codec_test.go:36]: Because while I was doing these changes, I was two commits behind the main. I was manually resolving conflicts by comparing changes. But by mistake, I might have forgotten.

removing it now
